### ISSUE TITLE
Classes aliased with class_alias are not considered equivalent for return type checks

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#1113]
https://getpsalm.org/r/3966c93f32
```php
<?php
  class C {}
  class_alias(C::class, 'Aliased', false);

  function f(): C {
    return new C;
  }

  function getAliased(): Aliased {
    return f();
  }
```
Since `Aliased` is the same as `C`, I do not expect Psalm to complain here. In synthetic cases (on getpsalm and locallly) I get UndefinedClass + InvalidReturnType + InvalidReturnStatement, however in real-life scenario I'm getting:
```
MoreSpecificReturnType: The declared return type 'Twig\TemplateWrapper' for ... is more specific than the inferred return type 'Twig_TemplateWrapper'                                                                                                     
LessSpecificReturnStatement: The type 'Twig_TemplateWrapper' is more general than the declared return type 'Twig\TemplateWrapper' for ... 
```

**Comment by muglug** [comment]
Yeah currently Psalm only considers `class_alias` when found in bootstrap/autoloader files. I'll make it compatible within files.

**Comment by weirdan** [comment]
Issue is still reported, but it appears Psalm is not _completely_ wrong. Live sample here: https://github.com/weirdan/psalm-class-alias-compatibility

Basically Twig is a mess.
